.github/workflows/scala.yml

```diff
@@ -1,17 +1,88 @@
-name: Scala CI
+name: Scala CI & Release
 
-on: [push, pull_request]
+on:
+  push:
+    branches: [ '*' ] # Trigger on all branches
+    tags:
+      - 'v*' # Trigger on version tags like v1.0.0
+  pull_request:
+    branches: [ master ] # Keep PR trigger for master
+  workflow_dispatch:
 
 jobs:
-  build:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 # Required for sbt-dynver to access Git history
+    - name: Set up JDK 11
+      uses: actions/setup-java@v4
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: 'sbt'
+    - name: Setup sbt
+      uses: coursier/setup-action@v1
+      with:
+        apps: sbt
+    - name: Run tests
+      run: sbt +test
 
+  publish-snapshot:
```

Comment:
Publish snapshot should automatically publish on github, and only manually on sonatype